src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -224,4 +226,17 @@ class LibraryService (protected val argUserInfo: UserInfo,
       case e: FireCloudException => Future(RequestCompleteWithErrorReport(BadRequest, s"suggestions not available for field %s".format(field)))
     }
   }
+
+  def searchOrspId(orspId: String): Future[PerRequestMessage] = {
+    // We need this implicit to unmarshal the consent in the response
+    import ModelJsonProtocol.impDuosConsent
+    ontologyDAO.orspIdSearch(userInfo, orspId) map { RequestComplete(_) } recoverWith {
```

Comment:
instead of using `recoverWith` and having your exception cases return `Future(RequestComplete...)`, you could use `recover` and have your exception cases return `RequestComplete...`

(yes I am also guilty of using recoverWith when I should use recover.